common4j/src/main/com/microsoft/identity/common/java/opentelemetry/IFidoTelemetryHelper.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.opentelemetry;
+
+import lombok.NonNull;
+
+public interface IFidoTelemetryHelper extends ITelemetryHelper {
+    /**
+     * Sets attribute indicating the type of FIDO challenge received from the server.
+     * @param challengeName name of the FidoChallenge class.
+     */
+    void setFidoChallenge(@NonNull final String challengeName);
+
+    /**
+     * Sets attribute indicating the type of FIDO challenge handler handling the current FIDO operation.
+     * @param challengeHandlerName name of the FidoChallengeHandler class.
+     */
+    void setFidoChallengeHandler(@NonNull final String challengeHandlerName);
```

Comment:
Usually interfaces are not used for declaring setter methods. Is there a specific reason these setters cannot be declared directly on the class? Do you every imagine having different implementations that would need to set these values _differently_?

Reply:
For the Fido/Passkey feature, not really (see response to Mohit's comment). For CBA, a helper was needed because there were multiple classes/times where attributes were being set. Though maybe it didn't have to be an interface. That's something I can address in a separate PR for CBA, but for this one, based on the comments received, I'm planning on removing these helper and directly call methods on the Otel span objects.

Reply:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2183/commits/7aedde14b813a93bf9a198865da859b68af12806